src/Aspire.Hosting/DistributedApplicationBuilder.cs

```diff
@@ -80,12 +78,40 @@ public DistributedApplicationBuilder(DistributedApplicationOptions options)
 
         AppHostDirectory = options.ProjectDirectory ?? _innerBuilder.Environment.ContentRootPath;
 
+        // Set configuration
+        ConfigurePublishingOptions(options);
         _innerBuilder.Configuration.AddInMemoryCollection(new Dictionary<string, string?>
         {
             // Make the app host directory available to the application via configuration
             ["AppHost:Directory"] = AppHostDirectory
         });
 
+        if (!options.DisableDashboard && !IsDashboardUnsecured(_innerBuilder.Configuration))
```

Comment:
Maybe extract this into its own method as well?

Reply:
This is fine. They're separate concerns.